fix(ast): change `ImportExpression::attributes` to `options`

Boshen opened the pull request:
To align with ts-eslint

https://github.com/typescript-eslint/typescript-eslint/blob/9aa0b6b49807599b1c2b704b778df74c91084a7c/packages/ast-spec/src/expression/ImportExpression/spec.ts#L23

Boshen commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9665?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#9665** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9665?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9665?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#9664** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9664?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=9665).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
@overlookmotel simply changing this field breaks `raw_transfer`, am I missing something?

```
[tasks/ast_tools/src/generators/raw_transfer.rs:772:9] field_name = "arguments"

thread '<unnamed>' panicked at tasks/ast_tools/src/generators/raw_transfer.rs:773:87:
called `Option::unwrap()` on a `None` value
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

overlookmotel commented:
> @overlookmotel simply changing this field breaks `raw_transfer`, am I missing something?

Sorry. Not a very helpful error message!

Usually, just changing the field name is enough. But where there's a custom serializer (`#[estree(via = Converter)]`), you have to implement serialization/deserialization twice - once in Rust in `ESTree` impl on the "converter" type, and second time in JS in the `#[estree(raw_deser = "...")]` attribute on that same type. The fragment of JS in `raw_deser` attr gets inserted into the JS-side deserializer code.

This is the commit that fixes it: https://github.com/oxc-project/oxc/pull/9665/commits/2c4af1b1599eec3c1ecf1a8d22813420a3bd3562

The syntax of the `raw_deser` code is pretty dreadful and is going to be largely incomprehensible to anyone but me. I'll try to improve it.

Boshen commented:
### Merge activity

* **Mar 11, 6:22 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Boshen committed:
fix(ast): change `ImportExpression::attributes` to `options` (#9665)

To align with ts-eslint

https://github.com/typescript-eslint/typescript-eslint/blob/9aa0b6b49807599b1c2b704b778df74c91084a7c/packages/ast-spec/src/expression/ImportExpression/spec.ts#L23
